include/SDL3/SDL_gpu.h

```diff
@@ -704,21 +744,41 @@ typedef struct SDL_GpuTextureSamplerBinding
 	SDL_GpuSampler *sampler;
 } SDL_GpuTextureSamplerBinding;
 
-typedef struct SDL_GpuComputeBufferBinding
+typedef struct SDL_GpuStorageBufferBinding
 {
 	SDL_GpuBuffer *gpuBuffer;
 
     /* if SDL_TRUE, cycles the buffer if it is bound. */
 	SDL_bool cycle;
-} SDL_GpuComputeBufferBinding;
+} SDL_GpuStorageBufferBinding;
 
-typedef struct SDL_GpuComputeTextureBinding
+typedef struct SDL_GpuStorageTextureBinding
 {
 	SDL_GpuTextureSlice textureSlice;
 
     /* if SDL_TRUE, cycles the texture if the texture slice is bound. */
 	SDL_bool cycle;
-} SDL_GpuComputeTextureBinding;
+} SDL_GpuStorageTextureBinding;
+
+typedef struct SDL_GpuUniformBufferBinding
+{
+	SDL_GpuUniformBuffer *uniformBuffer;
+	Uint32 uniformDataSizeInBytes;
+} SDL_GpuUniformBufferBinding;
+
+typedef struct SDL_GpuShaderResourceBinding
+{
+	SDL_GpuShaderResourceType resourceType;
+
+	SDLGPUNAMELESS union {
```

Comment:
If we're unable to use nameless on the required compilers, `resource` is probably an okay name for this union